src/functionalTest/java/uk/gov/hmcts/reform/ccd/utils/ElasticSearchTestUtils.java

```diff
@@ -78,42 +71,33 @@ private void refreshAllIndexes(final Set<String> indexes) {
     }
 
     public List<String> getAllDocuments(final String indexName) throws IOException {
-        final SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder()
-                .query(QueryBuilders.matchAllQuery());
-        final SearchRequest searchRequest = new SearchRequest(indexName)
-                .source(searchSourceBuilder);
-
-        final SearchResponse searchResponse = elasticsearchClient.search(searchRequest, DEFAULT);
-
-        return Arrays.stream(searchResponse.getHits().getHits())
-                .filter(hit -> indexName.startsWith(hit.getIndex()))
-                .map(SearchHit::getId)
-                .collect(Collectors.toUnmodifiableList());
+        final SearchRequest searchRequest = SearchRequest.of(s -> s
+            .index(indexName)
+            .query(q -> q
+                .matchAll(m -> m)
+            )
+            .source(SourceConfig.of(sc -> sc.fetch(true)))
+        );
+
+        final SearchResponse<Map> searchResponse = elasticsearchClient.search(searchRequest, Map.class);
+
+        return searchResponse.hits().hits().stream()
+            .filter(hit -> indexName.startsWith(hit.index()))
+            .map(Hit::id)
+            .collect(Collectors.toUnmodifiableList());
     }
 
     public void resetIndices(final Set<String> caseTypes) throws Exception {
-        final BulkRequest bulkRequest = new BulkRequest()
-                .setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
-
-        caseTypes.forEach(ThrowingConsumer.unchecked(caseType -> {
-
+        for (String caseType : caseTypes) {
             final String indexName = getIndexName(caseType);
-            if (elasticsearchClient.indices().exists(new GetIndexRequest(indexName), DEFAULT)) {
+            if (elasticsearchClient.indices().exists(e -> e.index(indexName)).value()) {
                 final List<String> documents = getAllDocuments(indexName);
-                documents.forEach(documentId -> {
-                    final DeleteRequest deleteRequest = new DeleteRequest(indexName)
-                            .id(documentId)
-                            .type(parameterResolver.getCasesIndexType());
-                    bulkRequest.add(deleteRequest);
-                });
-            }
-        }));
-
-        if (bulkRequest.numberOfActions() > 0) {
-            final BulkResponse bulkResponse = elasticsearchClient.bulk(bulkRequest, DEFAULT);
-
-            if (bulkResponse.hasFailures()) {
-                throw new Exception("Errors resetting indices::: " + bulkResponse.buildFailureMessage());
+                for (String documentId : documents) {
+                    elasticsearchClient.delete(d -> d
```

Comment:
We can use deleteByQuery() instead, it will give us the option to verify if there are any errors and refresh the index